wrappers/src/marshalling.hpp

```diff
@@ -387,6 +387,33 @@ static inline realm_value_t to_capi(Mixed value)
     return val;
 }
 
+inline realm_value_t to_capi(object_store::Dictionary& dictionary, Mixed val)
+{
+    if (val.is_null()) {
+        return to_capi(std::move(val));
+    }
+
+    switch (val.get_type()) {
+    case type_Link:
+        if ((dictionary.get_type() & ~PropertyType::Flags) == PropertyType::Object) {
+            return to_capi(new Object(dictionary.get_realm(), ObjLink(dictionary.get_object_schema().table_key, val.get<ObjKey>())));
+        }
+
+        REALM_UNREACHABLE();
+    case type_TypedLink:
```

Comment:
I am not sure what is the difference between `Link` and `TypedLink`

Reply:
I can roughly answer to that. `Link` should refer to objects within the same table. This translates in the fact that `Link` objects don't have a index pointing to what table they are from. While `TypedLink` are object that carry with them information about the table they belong to. Hence the 2 aren't comparable because you can't know where one is coming from.
@nirinchev will correct my comment and/or expand on it, but this is the general difference.